src/figuresClasses.ts

```diff
@@ -1,11 +1,103 @@
-export interface Figure {}
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  protected sidesArray: number[] = [this.a, this.b, this.c];
+
+  protected longestSide: number = this.sidesArray.splice(
+    this.sidesArray.indexOf(Math.max(...this.sidesArray)),
+  )[0];
+
+  protected otherSidesSum: number = this.sidesArray.reduce(
+    (acc: number, sideLength: number) => acc + sideLength,
+    0,
+  );
+
+  protected semiperimeter: number = (this.longestSide + this.otherSidesSum) / 2;
 
-export class Triangle implements Figure {}
+  protected isSidesWrong: boolean = this.longestSide >= this.otherSidesSum;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    readonly shape: Shape = Shape.triangle,
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to define 'color', 'a', 'b', 'c', and 'shape' properties explicitly inside the constructor since they are already being created by TypeScript.